src/test/fuzz/script_sigcache.cpp

```diff
@@ -2,44 +2,35 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <chainparams.h>
-#include <key.h>
+#include <consensus/amount.h>
+#include <primitives/transaction.h>
 #include <pubkey.h>
+#include <script/interpreter.h>
 #include <script/sigcache.h>
+#include <span.h>
 #include <test/fuzz/FuzzedDataProvider.h>
 #include <test/fuzz/fuzz.h>
 #include <test/fuzz/util.h>
-#include <test/util/setup_common.h>
+#include <uint256.h>
 
-#include <cstdint>
+#include <cstddef>
 #include <optional>
-#include <string>
 #include <vector>
 
-namespace {
-const BasicTestingSetup* g_setup;
-SignatureCache* g_signature_cache;
-} // namespace
-
-void initialize_script_sigcache()
-{
-    static const auto testing_setup = MakeNoLogFileContext<>();
-    static SignatureCache signature_cache{DEFAULT_SIGNATURE_CACHE_BYTES};
-    g_setup = testing_setup.get();
-    g_signature_cache = &signature_cache;
-}
-
-FUZZ_TARGET(script_sigcache, .init = initialize_script_sigcache)
+FUZZ_TARGET(script_sigcache)
 {
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
 
+    const auto max_sigcache_bytes{fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, DEFAULT_SIGNATURE_CACHE_BYTES)};
+    SignatureCache signature_cache{max_sigcache_bytes};
```

Comment:
I think we still need to setup logging or this test will oom because this constructor logs.